cmake/podioMacros.cmake

```diff
@@ -129,21 +129,27 @@ endfunction()
 # this is essentially a no-op, and should not cause re-compilation.
 #---------------------------------------------------------------------------------------------------
 function(PODIO_GENERATE_DATAMODEL datamodel YAML_FILE RETURN_HEADERS RETURN_SOURCES)
-  CMAKE_PARSE_ARGUMENTS(ARG "" "OUTPUT_FOLDER" "IO_BACKEND_HANDLERS" ${ARGN})
+  CMAKE_PARSE_ARGUMENTS(ARG "" "OUTPUT_FOLDER;FORMAT" "IO_BACKEND_HANDLERS" ${ARGN})
   IF(NOT ARG_OUTPUT_FOLDER)
     SET(ARG_OUTPUT_FOLDER ${CMAKE_CURRENT_SOURCE_DIR})
   ENDIF()
   IF(NOT ARG_IO_BACKEND_HANDLERS)
     # At least build the ROOT selection.xml by default for now
     SET(ARG_IO_BACKEND_HANDLERS "ROOT")
   ENDIF()
+  IF ("${ARG_FORMAT}" STREQUAL "")
```

Comment:
Not giving an argument to FORMAT will produce an error in cmake, won't it?

Reply:
Just checked with cmake@3.16.3 (Ubuntu) and all of the following calls seem to work as intended

- Passing in the value of a variable (passing a "false-y" value here is the only way to turn disable auto formatting)
```cmake
PODIO_GENERATE_DATAMODEL(datamodel datalayout.yaml headers sources
  FORMAT ${FORMAT_GENCODE} IO_BACKEND_HANDLERS ${PODIO_IO_HANDLERS})
```

- Simply passing `FORMAT` without any additional argument (will fall back to the default `ON`)
```cmake
PODIO_GENERATE_DATAMODEL(datamodel datalayout.yaml headers sources
  FORMAT IO_BACKEND_HANDLERS ${PODIO_IO_HANDLERS})
```

Reply:
Ok, thanks for checking!